core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -2194,7 +2193,14 @@ private Expression maybeBox(Expression expression) {
     @Override Expression implementSafe(RexToLixTranslator translator,
         RexCall call, List<Expression> argValueList) {
       final Expression argValue = argValueList.get(0);
-      final UnaryExpression e = Expressions.makeUnary(expressionType, argValue);
+
+      final Expression e;
+      if (expressionType == ExpressionType.Negate && argValue.type == BigDecimal.class) {
+        e = Expressions.call(BuiltInMethod.DECIMAL_NEGATE.method, argValue);
```

Comment:
According to Julian's suggestion, we can use `Expressions.call(argValue, "negate")` directly here, and do not need extra functions.

Reply:
Yes, this may a simple and more efficient